app/src/main/java/org/simple/clinic/protocolv2/ProtocolRepository.kt

```diff
@@ -59,10 +65,99 @@ class ProtocolRepository @Inject constructor(
     return protocolDao.count().toObservable()
   }
 
-  private fun payloadToProtocolDrugWithDosage(payload: ProtocolPayload): ProtocolDrugsWithDosage {
-    return ProtocolDrugsWithDosage(
+  private fun payloadToProtocolAndDrugs(payload: ProtocolPayload): ProtocolAndProtocolDrugs {
+    return ProtocolAndProtocolDrugs(
         protocol = payload.toDatabaseModel(newStatus = SyncStatus.DONE),
-        drugs = payload.protocolDrugs?.map { it.toDatabaseModel() } ?: emptyList()
+        drugs = payload.protocolDrugs.orEmpty().map { it.toDatabaseModel() }
     )
   }
+
+  fun drugsForProtocol(protocolId: UUID): Observable<List<ProtocolDrugAndDosages>> {
+    val protocolDrugsWithConfig = protocolDrugsDao
+        .drugsForProtocolId(protocolId)
+        .toObservable()
+        .withLatestFrom(config.toObservable())
+        .replay()
+        .refCount()
+
+    val syncedDrugs = protocolDrugsWithConfig
+        .filter { (drugs, config) -> drugs.isNotEmpty() && config.isProtocolDrugSyncEnabled }
+        .map { (drugs) ->
+          val drugsGroupedByName = drugs.groupBy { it.name }
+          drugsGroupedByName.map { (drugName, protocolDrugs) -> ProtocolDrugAndDosages(drugName, protocolDrugs) }
+        }
+
+    val defaultDrugs = protocolDrugsWithConfig
+        .filter { (drugs, config) -> drugs.isEmpty() || config.isProtocolDrugSyncEnabled.not() }
+        .map { defaultProtocolDrugs() }
+
+    return syncedDrugs.mergeWith(defaultDrugs)
+  }
+
+  @VisibleForTesting
+  fun defaultProtocolDrugs(): List<ProtocolDrugAndDosages> {
+    val protocolDrug = { uuid: UUID, name: String, dosage: String ->
+      ProtocolDrug(
+          uuid = uuid,
+          protocolUuid = UUID.fromString("d486b15b-ac2a-4d2f-b87c-e1905d5ee791"),
+          name = name,
+          rxNormCode = "rxNormCode-1",
```

Comment:
We really should be getting the actual rx norm codes instead of using dummy ones.

Reply:
I will do this